ios/RN/RNCamera.m

```diff
@@ -641,6 +653,57 @@ - (void)updateWhiteBalance
     [device unlockForConfiguration];
 }
 
+- (void)applyCustomWhiteBalance {
+    AVCaptureDevice *device = [self.videoCaptureDeviceInput device];
+    if(device == nil) {
+        return;
+    }
+
+    NSError *error = nil;
+    if (![device lockForConfiguration:&error]) {
+        if (error) {
+            RCTLogError(@"%s: %@", __func__, error);
+        }
+        return;
+    }
+
+    if (![device isWhiteBalanceModeSupported:AVCaptureWhiteBalanceModeLocked]) {
+        RCTLogWarn(@"%s: locked whitebalance mode ist note supported. Fallback to continuous auto white balance mode", __func__);
+        [device setWhiteBalanceMode:AVCaptureWhiteBalanceModeContinuousAutoWhiteBalance];
+    } else {
+        AVCaptureWhiteBalanceTemperatureAndTintValues temperatureAndTint = {
+            .temperature = self.customWhiteBalanceSettings.temperature,
+            .tint = self.customWhiteBalanceSettings.tint,
+        };
+        AVCaptureWhiteBalanceGains rgbGains = [device deviceWhiteBalanceGainsForTemperatureAndTintValues:temperatureAndTint];
+        CGFloat redGain = rgbGains.redGain + self.customWhiteBalanceSettings.redGainOffset;
+        CGFloat greenGain = rgbGains.greenGain + self.customWhiteBalanceSettings.greenGainOffset;
+        CGFloat blueGain = rgbGains.blueGain + self.customWhiteBalanceSettings.blueGainOffset;
+        
+        rgbGains.redGain = MAX(1.0f, MIN(device.maxWhiteBalanceGain, redGain));
+        rgbGains.greenGain = MAX(1.0f, MIN(device.maxWhiteBalanceGain, greenGain));
+        rgbGains.blueGain = MAX(1.0f, MIN(device.maxWhiteBalanceGain, blueGain));
+        
+        __weak __typeof__(device) weakDevice = device;
+        if ([device lockForConfiguration:&error]) {
+            @try{
+                [device setWhiteBalanceModeLockedWithDeviceWhiteBalanceGains:rgbGains completionHandler:^(CMTime syncTime) {
+                    [weakDevice unlockForConfiguration];
+                }];
+            }
+            @catch(NSException *exception){
+                RCTLogError(@"Failed to set custom white balance: %@", exception);
+            }
+        } else {
+            if (error) {
+                RCTLogError(@"%s: %@", __func__, error);
```

Comment:
A very bad way to re-write this in  `javascript` would be creating a method/function and passing a function as parameter ... It would only be good as a temporary re-factor to start re-writing this logic

```javascript
function getDeviceAndHandleErrors(method) {
    const device = videoCaptureDeviceInput();
    const error = nil;

    if(device == nil){
        return;
    }

    // more error handling logic

    method()
  
    // more logic
}

function updateExposure() {
    // your logic here
}

getDeviceAndHandleErrors(updateExposure)
```

But I believe the error handling logic should be somehow share between different methods, as they all use

```
    if(device == nil) {
       return
    }
```

this logic should be moved up to the `videoCaptureDeviceInput` method so there is no need to repeat it and using `return`

Reply:
Yes, you are right. I made a new pull request for this: https://github.com/react-native-community/react-native-camera/pull/2775.

p.s. you are not disturbing me :)

Reply:
Thanks for the awesome new pull request. I would keep two separate pull request. 

1) PULL 2775: fixing the basic example for ios (I will test this pull request on android)
2) PULL 2774: adding a customWhiteBalance method 

It will also help future developers review the not-merge pull request to build similar functionalities... Thanks a lot for the 2 amazing pull requests.